File: tightbind/lovlap.f
      SUBROUTINE LOVLAP (STRAD,A,B,SK1,SK2,R,L1,L2,M1,N1,N2,MAX)
C
C  ********************************************************************
C  *                                                                  *
C  *     SUBROUTINE LOVLAP       CALLED FROM MOV                   *
C  *                                                                  *
C  *                                                                  *
C  *       LOVLAP   SUBROUTINE TO CALCULATE THE OVERLAP               *
C  *                COMPONENT INDEPENDENT OF THE ANGLE BETWEEN        *
C  *                THE ATOMS                                         *
C  *                                                                  *
C  *       SUBROUTINES USED:                                          *
C  *                                                                  *
C  *             NONE                                                 *
C  *                                                                  *
C  *                                                                  *
C  *       ORIGIN LOST IN ANTIQUITY                                   *
C
C    modified by greg in modernity (august 1993) so that it doesn't use that
C         damn common block any more.
C
C  *                                                                  *
C  ********************************************************************
C
      IMPLICIT REAL*8(A-H,O-Z)
C
      DIMENSION A(30),B(30)
      DIMENSION FACT(25)
      DIMENSION BINCOE(8,8)
C
C      COMMON/LOCLAP/SK1,SK2,R,L1,L2,M1,N1,N2,MAX
C
      DATA BINCOE/8*1.D0,   0.D0,1.D0,2.D0,3.D0,4.D0,5.D0,6.D0,7.D0,
     1 2*0.D0,1.D0,3.D0,6.D0,10.D0,15.D0,21.D0,   3*0.D0,1.D0,4.D0,
     2 10.D0,20.D0,35.D0,   4*0.D0,1.D0,5.D0,15.D0,35.D0,   5*0.D0,
     3 1.D0,6.D0,21.D0,   6*0.D0,1.D0,7.D0,   7*0.D0,1.D0/
C

C      write (*,*) 'Lovlap: ',sk1,sk2,r,l1,l2,m1,n1,n2,max

      MAXXA=1
      MAXXB=1
      MAXXC=1
      FACT(1)=1.D0
      DO 5 I=2,25
 5    FACT(I)=FACT(I-1)*DBLE(FLOAT(I-1))
      M2=M1
      STRAD=0.D0
      RHOA=R*SK1
      RHOB=R*SK2
      RHOAP=RHOA**N1
      RHOAP=RHOAP*RHOAP
      RHOAP=RHOAP*RHOA
      RHOAB=RHOB**N2
      RHOAB=RHOAB*RHOAB
      RHOAB=RHOAB*RHOB
      RHOPO=RHOAP*RHOAB
      TERMA=0.5D0**(L1+L2+1)*DSQRT(DBLE(FLOAT((L1+L1+1)*(L2+L2+1)))*
     1 FACT(L1-M1+1)*FACT(L2-M1+1)/(FACT(N1+N1+1)*FACT(N2+N2+1)*
     2 FACT(L1+M1+1)*FACT(L2+M1+1))*RHOPO)
      JEND=1+((L1-M1)/2)
      KEND=1+((L2-M2)/2)
      IEB=M1+1
      DO 50 J=1,JEND
      JU=J-1
      IAB=N1-L1+JU+JU+1
      ICB=L1-M1-JU-JU+1
      CON1=FACT(L1+L1-JU-JU+1)/(FACT(L1-M1-JU-JU+1)*FACT(JU+1)*
     1 FACT(L1-JU+1))
      DO 50 K=1,KEND
      KU=K-1
      CON12=CON1*FACT(L2+L2-KU-KU+1)/(FACT(L2-M2-KU-KU+1)*FACT(KU+1)*
     1 FACT(L2-KU+1))
      IEV=JU+KU+L2
      IF(2*(IEV/2).NE.IEV) CON12=-CON12
      IBB=N2-L2+KU+KU+1
      IDB=L2-M2-KU-KU+1
      VALUE=0.D0
      DO 90 I6=1,IEB
      DO 90 I5=1,IEB
      VALUE1=BINCOE(IEB,I6)*BINCOE(IEB,I5)
      IEV=I5+I6
      IF(2*(IEV/2).NE.IEV) VALUE1=-VALUE1
      DO 90 I4=1,IDB
      VALUE1=-VALUE1
      VALUE2=BINCOE(IDB,I4)*VALUE1
      DO 90 I3=1,ICB
      VALUE3=BINCOE(ICB,I3)*VALUE2
      DO 90 I2=1,IBB
      VALUE3=-VALUE3
      VALUE4=BINCOE(IBB,I2)*VALUE3
      DO 90 I1=1,IAB
      TERM=VALUE4*BINCOE(IAB,I1)
      IR=I1+I2+IEB+IEB-I6-I6-I3+IDB-I4+ICB-1
      IP=IAB-I1+IBB-I2+IEB+IEB-I5-I5+ICB-I3+IDB-I4+1
90    VALUE=VALUE+A(IP)*B(IR)*TERM
50    STRAD=STRAD+VALUE*CON12
      STRAD=STRAD*TERMA

      RETURN
      END

File: tightbind/lovlap.c
/*
Produced by gmFortran V30.59(10/26/17) on 9/17/18 at 9:45:37
*/
#define LPROTOTYPE
#include "fortran.h"
/*
*/
void lovlap(double *strad,double *a,double *b,double *sk1,double *sk2,double *r,
    int *l1,int *l2,int *m1,int *n1,int *n2,int *max)
{
/*
  ********************************************************************
  *                                                                  *
  *     SUBROUTINE LOVLAP       CALLED FROM MOV                   *
  *                                                                  *
  *                                                                  *
  *       LOVLAP   SUBROUTINE TO CALCULATE THE OVERLAP               *
  *                COMPONENT INDEPENDENT OF THE ANGLE BETWEEN        *
  *                THE ATOMS                                         *
  *                                                                  *
  *       SUBROUTINES USED:                                          *
  *                                                                  *
  *             NONE                                                 *
  *                                                                  *
  *                                                                  *
  *       ORIGIN LOST IN ANTIQUITY                                   *

    modified by greg in modernity (august 1993) so that it doesn't use that
         damn common block any more.

  *                                                                  *
  ********************************************************************

*/
/*
      COMMON/LOCLAP/SK1,SK2,R,L1,L2,M1,N1,N2,MAX

*/
static double bincoe[8][8] = {
    1.e0,1.e0,1.e0,1.e0,1.e0,1.e0,1.e0,1.e0,0.e0,1.e0,2.e0,3.e0,4.e0,5.e0,6.e0,
    7.e0,0.e0,0.e0,1.e0,3.e0,6.e0,10.e0,15.e0,21.e0,0.e0,0.e0,0.e0,1.e0,4.e0,
    10.e0,20.e0,35.e0,0.e0,0.e0,0.e0,0.e0,1.e0,5.e0,15.e0,35.e0,0.e0,0.e0,0.e0,
    0.e0,0.e0,1.e0,6.e0,21.e0,0.e0,0.e0,0.e0,0.e0,0.e0,0.e0,1.e0,7.e0,0.e0,0.e0,
    0.e0,0.e0,0.e0,0.e0,0.e0,1.e0
};
static double fact[25];
static int maxxa,maxxb,maxxc,i,m2;
static double rhoa,rhob,rhoap,rhoab,rhopo,terma;
static int jend,kend,ieb,j,ju,iab,icb;
static double con1;
static int k,ku;
static double con12;
static int iev,ibb,idb;
static double value;
static int i6,i5;
static double value1;
static int i4;
static double value2;
static int i3;
static double value3;
static int i2;
static double value4;
static int i1;
static double term;
static int ir,ip;
/*
      write (*,*) 'Lovlap: ',sk1,sk2,r,l1,l2,m1,n1,n2,max
*/
    maxxa = 1;
    maxxb = 1;
    maxxc = 1;
    fact[0] = 1.e0;
    for(i=1; i<25; i++) {
        fact[i] = fact[i-1]*(double)(float)i;
    }
    m2 = *m1;
    *strad = 0.e0;
    rhoa = *r**sk1;
    rhob = *r**sk2;
    rhoap = pow(rhoa,(double)*n1);
    rhoap = rhoap*rhoap;
    rhoap = rhoap*rhoa;
    rhoab = pow(rhob,(double)*n2);
    rhoab = rhoab*rhoab;
    rhoab = rhoab*rhob;
    rhopo = rhoap*rhoab;
    terma = pow(0.5e0,(double)(*l1+*l2+1))*sqrt((double)(float)((*l1+*l1+1)*(*
      l2+*l2+1))*fact[*l1-*m1]*fact[*l2-*m1]/(fact[*n1+*n1]*fact[*n2+*n2]*fact[*
      l1+*m1]*fact[*l2+*m1])*rhopo);
    jend = 1+(*l1-*m1)/2;
    kend = 1+(*l2-m2)/2;
    ieb = *m1+1;
    for(j=0; j<jend; j++) {
        ju = j;
        iab = *n1-*l1+ju+ju+1;
        icb = *l1-*m1-ju-ju+1;
        con1 = fact[*l1+*l1-ju-ju]/(fact[*l1-*m1-ju-ju]*fact[ju]*fact[*l1-ju]);
        for(k=0; k<kend; k++) {
            ku = k;
            con12 = con1*fact[*l2+*l2-ku-ku]/(fact[*l2-m2-ku-ku]*fact[ku]*fact[*
              l2-ku]);
            iev = ju+ku+*l2;
            if(2*(iev/2) != iev) con12 = -con12;
            ibb = *n2-*l2+ku+ku+1;
            idb = *l2-m2-ku-ku+1;
            value = 0.e0;
            for(i6=1; i6<=ieb; i6++) {
                for(i5=1; i5<=ieb; i5++) {
                    value1 = bincoe[i6-1][ieb-1]*bincoe[i5-1][ieb-1];
                    iev = i5+i6;
                    if(2*(iev/2) != iev) value1 = -value1;
                    for(i4=1; i4<=idb; i4++) {
                        value1 = -value1;
                        value2 = bincoe[i4-1][idb-1]*value1;
                        for(i3=1; i3<=icb; i3++) {
                            value3 = bincoe[i3-1][icb-1]*value2;
                            for(i2=1; i2<=ibb; i2++) {
                                value3 = -value3;
                                value4 = bincoe[i2-1][ibb-1]*value3;
                                for(i1=1; i1<=iab; i1++) {
                                    term = value4*bincoe[i1-1][iab-1];
                                    ir = i1+i2+ieb+ieb-i6-i6-i3+idb-i4+icb-1;
                                    ip = iab-i1+ibb-i2+ieb+ieb-i5-i5+icb-i3+idb-
                                      i4+1;
                                    value = value+a[ip-1]*b[ir-1]*term;
                                }
                            }
                        }
                    }
                }
            }
            *strad = *strad+value*con12;
        }
    }
    *strad = *strad*terma;
    return;
}
